src/main/java/core/basesyntax/service/impl/FileWriterServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.WriterService;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+
+public class FileWriterServiceImpl implements WriterService {
+    private static final String CANT_WRITE = "Can't write to file. ";
+    private static final String COMMA = ",";
+    private static final String PREDEFINED_LINE = "fruit,quantity";
+    private final BufferedWriter bufferedWriter;
+
+    public FileWriterServiceImpl(BufferedWriter bufferedWriter) {
+        this.bufferedWriter = bufferedWriter;
+    }
```

Comment:
you can initialize bufferedWriter with new BufferedWriter() constructor without parameters

Reply:
if you use operator new you'll have strong binding in the class